report/string_set.go

```diff
@@ -54,33 +54,35 @@ func (s StringSet) Intersection(b StringSet) StringSet {
 // Add adds the strings to the StringSet. Add is the only valid way to grow a
 // StringSet. Add returns the StringSet to enable chaining.
 func (s StringSet) Add(strs ...string) StringSet {
+	r := s.Copy()
```

Comment:
@rade Actually, I think we should make it immutable until we confirm that it's completely safe. There are some non-obvious use-cases like  IDList.Add() and Controls.Add() See https://github.com/weaveworks/scope/commit/94347b9c5c087f16ed472fdc542eb4e749291030
 
I possibility is to add Unsafe variants, but I don't want to do that until I confirm there's a negligible gain.